component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java

```diff
@@ -188,13 +188,21 @@ public static ProfileEntity buildEntityProfile(Profile profile, String restPath,
     userEntity.setAvatar(profile.getAvatarUrl());
     userEntity.setBanner(profile.getBannerUrl());
     try {
+      boolean isExternal = profile.getProperty(Profile.EXTERNAL) != null && (profile.getProperty(Profile.EXTERNAL)).equals("true");
       OrganizationService organizationService = getOrganizationService();
       User user = organizationService.getUserHandler().findUserByName(userEntity.getUsername());
       if (user != null) {
         userEntity.setIsInternal(user.isInternalStore());
+        if (isExternal && user.getLastLoginTime().equals(user.getCreatedDate())) {
+          userEntity.setLastConnexion("neverConnected");
+        } else if (user.getLastLoginTime().equals(user.getCreatedDate())) {
+          userEntity.setLastConnexion("neverEnrolled");
+        } else {
+          userEntity.setLastConnexion(String.valueOf(user.getLastLoginTime().getTime()));
+        }
```

Comment:
we should not have 3 states here, only 2 : the date if exist, null else. 
And at display, we will check the enrollDate (that we dont have for the moment)